contracts/modules/TransferManager/MaximumVolumeTransferManager.sol

```diff
@@ -0,0 +1,307 @@
+    * @param _endTimes timestamps representing when these maximum volume restrictions end
+    * @param _rollingPeriodIntervals interval periods representing one of five RollingIntervals
+    */
+    function addMaxVolumeRestricionsMulti(
+        uint256[] _maximumVolumes, 
+        uint256[] _startTimes, 
+        uint256[] _endTimes, 
+        uint256[] _rollingPeriodIntervals) 
+    public withPerm(ADMIN) {
+        require(_maximumVolumes.length >= 1, "No restrictions provided");
+        require(
+            _maximumVolumes.length == _startTimes.length &&
+            _maximumVolumes.length == _endTimes.length &&
+            _maximumVolumes.length == _rollingPeriodIntervals.length,
+            "Input array length mismatch"
+        );
+
+        for(uint256 i = 0; i < _maximumVolumes.length; i++){
+            _addMaxVolumeRestricion(
+                _maximumVolumes[i],
+                _startTimes[i],
+                _endTimes[i],
+                _rollingPeriodIntervals[i]
+            );
+        }
+    }
+
+    /**
+    * @notice allows admin to modify a maximum volume transfer restriction
+    * @param _index representing the array index of the maximum volume transfer restriction
+    * @param _maximumVolume maximum token volume that can be transfered
+    * @param _startTime timestamp representing when this maximum volume restriction starts
+    * @param _endTime timestamp representing when this maximum volume restriction ends
+    * @param _rollingPeriodInterval interval period representing one of five RollingIntervals
+    */
+    function modifyMaximumTransferRestriction(
+        uint256 _index, 
+        uint256 _maximumVolume, 
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriodInterval) 
+    public withPerm(ADMIN) {
+        require(_index < maximumVolumeRestrictions.length, "invalid index");
+        require(now < maximumVolumeRestrictions[_index].endTime, "restriction is no longer active");
+
+        if(_startTime == 0) {
+            _startTime = now;
+        }
+
+        _checkRestrictionParams(_startTime, _endTime, _rollingPeriodInterval);
+
+        MaximumVolumeRestriction memory restriction = MaximumVolumeRestriction(
+            _maximumVolume, 
+            _startTime, 
+            _endTime, 
+            RollingIntervals(_rollingPeriodInterval)
+        );
+        maximumVolumeRestrictions[_index] = restriction;
+
+        emit ModifyMaximumVolumeRestriction(
+            _index,
+            now, 
+            _maximumVolume,
+            _startTime,
+            _endTime,
+            _rollingPeriodInterval
+        );
+    }
+
+    /**
+    * @notice allows admin to delete a maximum volume transfer restriction
+    * @param _index representing the array index of the maximum volume transfer restriction
+    */
+    function removeMaximumTransferRestriction(uint256 _index) public withPerm(ADMIN){
```

Comment:
Can you please add a batch/multi versions of this.

Reply:
@satyamakgec  @maxsam4 Currently, removing a single array element will almost always change the order of restrictions in the array( except when the restriction to be deleted in the last one). This could render some of the indices provided unusable and therefore cause the function to revert.

I could write an implementation which would preserve the order of restrictions by shifting the elements instead of swapping them(as implemented now) but doing so would make the function extremely costly in terms of gas. 
```
string element = myArray[index];
for (uint i = index; i < myArray.length - 1; i++) {
  myArray[index] = myArray[index + 1];
}
myArray.length--;
```

Please advise on how to proceed with this.

Reply:
@vyomshm Can you update the indices as well when deleting an array element?